app/code/Magento/SalesRule/Model/Coupon/UpdateCouponUsages.php

```diff
@@ -38,21 +39,29 @@ class UpdateCouponUsages
     private $couponUsage;
 
     /**
-     * @param RuleFactory $ruleFactory
+     * @var Customer
+     */
+    private $customerRuleUsage;
+
+    /**
+     * @param RuleFactory     $ruleFactory
      * @param CustomerFactory $ruleCustomerFactory
-     * @param Coupon $coupon
-     * @param Usage $couponUsage
+     * @param Coupon          $coupon
+     * @param Usage           $couponUsage
+     * @param Customer        $customerRuleUsage
      */
     public function __construct(
-        RuleFactory $ruleFactory,
+        RuleFactory     $ruleFactory,
         CustomerFactory $ruleCustomerFactory,
-        Coupon $coupon,
-        Usage $couponUsage
+        Coupon          $coupon,
+        Usage           $couponUsage,
+        Customer        $customerRuleUsage
```

Comment:
Please, inject new dependencies according to our [Backward compatible development guide](https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/)

Reply:
New Dependencies has injected. kindly check and let me know